ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplUtils.java

```diff
@@ -55,4 +106,18 @@
     }
     return TaskFactory.get(new DDLWork(new HashSet<>(), new HashSet<>(), alterTblDesc), conf);
   }
+
+  public static boolean replCkptStatus(String dbName, Map<String, String> props, String dumpRoot)
+          throws InvalidOperationException {
+    // If ckpt property not set or empty means, bootstrap is not run on this object.
+    if ((props != null) && props.containsKey(REPL_CHECKPOINT_KEY) && !props.get(REPL_CHECKPOINT_KEY).isEmpty()) {
+      if (props.get(REPL_CHECKPOINT_KEY).equals(dumpRoot)) {
```

Comment:
props.get(REPL_CHECKPOINT_KEY) is called 3 times ..can be cached to a local variable

Reply:
It is just a hash map and should be fast. Also, it is accessed 2 times in one flow.